examples/cis2-multi-royalties/src/lib.rs

```diff
@@ -0,0 +1,1547 @@
+        let is_operator = host.state().is_operator(&ADDRESS_1, &ADDRESS_0);
+        claim!(is_operator, "Account should be an operator");
+
+        // Checking that `ADDRESS_1` is an operator in the query response of the
+        // `contract_operator_of` function as well.
+        // Setup parameter.
+        let operator_of_query = OperatorOfQuery {
+            address: ADDRESS_1,
+            owner:   ADDRESS_0,
+        };
+
+        let operator_of_query_vector = OperatorOfQueryParams {
+            queries: vec![operator_of_query],
+        };
+        let parameter_bytes = to_bytes(&operator_of_query_vector);
+
+        ctx.set_parameter(&parameter_bytes);
+
+        // Checking the return value of the `contract_operator_of` function
+        let result: ContractResult<OperatorOfQueryResponse> = contract_operator_of(&ctx, &host);
+
+        claim_eq!(
+            result.expect_report("Failed getting result value").0,
+            [true],
+            "Account should be an operator in the query response"
+        );
+
+        // Check the logs.
+        claim_eq!(logger.logs.len(), 1, "One event should be logged");
+        claim_eq!(
+            logger.logs[0],
+            to_bytes(&Cis2Event::<ContractTokenId, ContractTokenAmount>::UpdateOperator(
+                UpdateOperatorEvent {
+                    owner:    ADDRESS_0,
+                    operator: ADDRESS_1,
+                    update:   OperatorUpdate::Add,
+                }
+            )),
+            "Incorrect event emitted"
+        )
+    }
+
+    // test royalties are initialised properly
+    #[concordium_test]
+    fn test_royalties_setup() {
+        // Setup the context
+        let ctx = TestReceiveContext::empty();
+        let mut builder1 = TestStateBuilder::new();
+        let state1 = initial_state(&mut builder1);
+        let host = TestHost::new(state1, builder1);
+        let royalty_paid =
+            contract_pays_royalties(&ctx, &host).expect_report("Invoke should succeed");
+        claim_eq!(royalty_paid, false, "Royalty incorrectly initialised");
+
+        let mut builder2 = TestStateBuilder::new();
+        let mut state2 = initial_state(&mut builder2);
+        state2.pay_royalty = true;
+        let host = TestHost::new(state2, builder2);
+        let royalty_paid =
+            contract_pays_royalties(&ctx, &host).expect_report("Invoke should succeed");
+        claim_eq!(royalty_paid, true, "Royalty incorrectly initialised");
+    }
+
+    // test royalties are initialised properly
+    #[concordium_test]
+    fn test_royalties() {
+        let ctx = TestReceiveContext::empty();
+        let mut builder = TestStateBuilder::new();
+        let state = initial_state(&mut builder);
+        let host = TestHost::new(state, builder);
+        let royalty_paid =
+            contract_pays_royalties(&ctx, &host).expect_report("Invoke should succeed");
+        claim_eq!(royalty_paid, false, "Royalty incorrectly initialised");
+    }
```

Comment:
This seems to test the same thing as line 1296-1303. Consider removing or clarifying what in addition you are testing here.

Reply:
Have refactored the code to make it clearer that it is testing a positive and negative case and removed duplicated test